### PR TITLE
initial commit

### PR DESCRIPTION
**Pull request by lhein**
Signed-off-by: Lars Heinemann <lhein.smx@gmail.com>

### CODE REVIEW

File: src/DataVirtNodeProvider.ts [hunk 1]
```diff
@@ -0,0 +1,444 @@
...
+		for (let node of oldNodes) {
+			if (node.label === newNode.label) {
+				return true;
+			}
+		}
+		return false;
+	}
+
+	// process the JSON we get back from the kube rest API
+	processDataVirtYAML(fullPath: string, name: string, yaml : IDVConfig): void {
```
Comment (apupier) [inline src/DataVirtNodeProvider.ts]: what about naming parameter yaml as dvConfig or datavirtConfig? When reading code, I was wondering how a yaml can have specific attributes kind or metadata.

File: package.json [hunk 2]
```diff
@@ -0,0 +1,185 @@
+{
+	"name": "datavirt-tools",
+	"displayName": "DataVirt Tooling by Red Hat",
+	"description": "",
+	"version": "0.0.1",
+	"publisher": "redhat",
+	"icon": "icons/icon128.png",
```
Comment (apupier) [inline package.json]: the icon is missing from the PR

File: package.json [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+	},
+	"activationEvents": [
+		"*"
+	],
+	"scripts": {
+		"vscode:prepublish": "tsc -p ./",
+		"compile": "tsc -p ./",
+		"watch": "tsc -w -p ./",
+		"clean": "rm -rf out || rmdir out /s /q",
+		"test": "node ./out/test/runTest.js",
```
Comment (apupier) [inline package.json]: runTest.js doesn't exist

File: package.json [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+		"update-deps": "npm update",
+		"ui-test": "npm run compile && extest setup-and-run out/ui-test/all.js",
```
Comment (apupier) [inline package.json]: all.js doesn't exist

File: Jenkinsfile [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+		sh "npm install --ignore-scripts"
+		sh "npm install"
+		sh "npm run vscode:prepublish"
+	}
+
+	withEnv(['JUNIT_REPORT_PATH=report.xml']) {
+        stage('Test') {
+    		wrap([$class: 'Xvnc']) {
+    			sh "npm test --silent"
+    			junit 'report.xml'
```
Comment (apupier) [inline Jenkinsfile]: currently there is no test so it will fail

File: Jenkinsfile [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+		env.PATH="${env.PATH}:${nodeHome}/bin"
+		sh "npm install -g typescript vsce"
+	}
+
+	stage('Build') {
+		env.JAVA_HOME="${tool 'openjdk-1.8'}"
+		env.PATH="${env.JAVA_HOME}/bin:${env.PATH}"
+		sh "java -version"
+		
+		sh "npm install --ignore-scripts"
```
Comment (apupier) [inline Jenkinsfile]: there is no script for the build, so we can remove this line

**Comment by apupier** [comment]
argh submitted too fast, not finished review

**Comment by apupier** [comment]
but already merged... so we can create issues for my feedback

**Comment by lhein** [comment]
Yes, this PR was meant as an initial commit. I did not expect a review at that point. Please raise issues for features mainly. Its currently still a work in progress and I even forgot to submit some files.

**Pull request merged**